File: C_submissions/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    float L = (float) count_letters(text) / (float) count_words(text) * 100;
    float S = (float) count_sentences(text) / (float) count_words(text) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;

    int roundedIndex = round(index);
    if (roundedIndex > 16)
    {
        printf("Grade 16+\n");
    }
    else if (roundedIndex < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", roundedIndex);
    }
}

int count_letters(string text)
{
    int lengthText = strlen(text);

    char c;
    int numberOfLetters = 0;
    for (int i = 0; i < lengthText; i++)
    {
        c = tolower(text[i]);
        if (c >= 'a' && c <= 'z')
        {
            numberOfLetters++;
        }
    }

    return numberOfLetters;
}

int count_words(string text)
{
    int lengthText = strlen(text);

    int numberOfWords = 0;

    for (int i = 0; i <= lengthText; i++)
    {
        if ((text[i] == ' ' && text[i - 1] != ' ') || text[i] == '\0')
        {
            numberOfWords++;
        }
    }

    return numberOfWords;
}

int count_sentences(string text)
{
    int lengthText = strlen(text);

    int numberOfSentences = 0;

    for (int i = 0; i < lengthText; i++)
    {
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            numberOfSentences++;
        }
    }

    return numberOfSentences;
}

File: PYTHON_submissions/sentimental-readability/readability.py
from cs50 import get_string
import re


def main():
    text = get_string("Text: ")

    avg_letters = count_letters(text) / count_words(text) * 100
    avg_sentences = count_sentences(text) / count_words(text) * 100

    index = round(0.0588 * avg_letters - 0.296 * avg_sentences - 15.8)

    if index > 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


def count_letters(text):

    text_lower = text.lower()

    letters = 0
    for character in text_lower:
        if ord(character) >= ord("a") and ord(character) <= ord("z"):
            letters += 1
    return letters


def count_words(text):

    arr_text = text.split(" ")
    len_words = len(arr_text)

    return len_words


def count_sentences(text):

    sentences = re.findall(r"[.!?]+", text)
    len_sentences = len(sentences)

    return len_sentences


main()
